tests/subsys/suit/unit/suit_decrypt_filter/src/main.c

```diff
@@ -0,0 +1,545 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <suit_decrypt_filter.h>
+#include <suit_mci.h>
+#include <zephyr/fff.h>
+#include <zephyr/ztest.h>
+#include <psa/crypto.h>
+#include <mocks.h>
+
+#define KEY_ID_FWENC_APPLICATION_GEN1 0x40022000
+#define ENC_INFO_DEFAULT_INIT \
+	{ \
+		.enc_alg_id = suit_cose_aes256_gcm, \
+		.IV = { \
+				 .value = iv_direct, \
+				 .len = sizeof(iv_direct), \
+			  }, \
+		.aad = { \
+				  .value = aad, \
+				  .len = sizeof(aad), \
+			   }, \
+		.kw_alg_id = suit_cose_direct, \
+		.kw_key.direct = {.key_id = {.value = cek_key_id_cbor, \
+		.len = sizeof(cek_key_id_cbor)},} \
+	}
+/**
+ * Encryption without wrapping CEK achieved by running:
+ *
+ * echo "This is a sample plaintext for testing the decryption filter" > plaintext.txt
+ * nrfkms encrypt -k  TEST_AES_KEY -c test -f plaintext.txt --aad "sample aad" --format native
+ *
+ * Ciphertext and NONCE (IV) taken from the encrypted_data_using_TEST_AES_KEY-test.bin file,
+ * which is in format |nonce (12 bytes)|tag (16 bytes)|ciphertext|
+ */
+
+static const uint8_t ciphertext_direct[] = {
+	/* tag (16 bytes) */
+	0x4d, 0x21, 0x30, 0xb7, 0xce, 0x8a, 0xd6, 0x00, 0xe4, 0x04, 0xbb, 0x32,
+	0x72, 0x7a, 0xbb, 0x7c,
+	/* ciphertext */
+	0xf0, 0x72, 0xdb, 0x63, 0x03, 0xdd, 0x24, 0x69,
+	0xd4, 0xbf, 0xd7, 0xa0, 0xec, 0xfa, 0x66, 0x58, 0x95, 0x2b, 0xc1, 0xc2,
+	0x9d, 0x82, 0x02, 0x1a, 0xd7, 0x5b, 0xc0, 0x01, 0xce, 0x0b, 0x79, 0x53,
+	0xe7, 0xdb, 0x0d, 0x35, 0xab, 0xef, 0x81, 0xc8, 0x68, 0xc5, 0xa7, 0x22,
+	0x90, 0xea, 0xd0, 0x7f, 0x36, 0xed, 0x14, 0xbe, 0x30, 0xf2, 0x81, 0x56,
+	0x7e, 0x2e, 0x5f, 0xd8, 0x7c,
+};
+
+
+static const uint8_t iv_direct[] = {
+	0x60, 0x90, 0x6d, 0xb2, 0xfe, 0xc3, 0xc8, 0x5a, 0xf0, 0x28, 0xb1, 0xb6,
+};
+
+static const suit_manifest_class_id_t sample_class_id = {
+	{0x5b, 0x46, 0x9f, 0xd1, 0x90, 0xee, 0x53, 0x9c, 0xa3, 0x18, 0x68, 0x1b, 0x03, 0x69, 0x5e,
+	 0x36}
+};
+
+static const uint8_t cek_key_id_cbor[] = {
+		0x1A, 0x40, 0x02, 0x20, 0x00,
+	};
+
+struct suit_decrypt_filter_tests_fixture {
+	struct stream_sink dec_sink;
```

Comment:
Could you change the names to out_sink/in_sink so that is clearer which one is which? Similarly as you've changed in the source file